### PR TITLE
Allow RestEasy JAXB collection serialization in native mode

### LINKED ISSUE #11806: Serialization of RestEasy JAXB wrapped collections fails in native mode
**Describe the bug**

When using the `org.jboss.resteasy.annotations.providers.jaxb.Wrapped` annotation in JAX-RS resource to force a specific Collection wrapping XML tag, the collection it not serialized in native mode. However it works as expected in JVM mode.

I use the following code:
```java
    @GET
    @Produces({MediaType.APPLICATION_JSON, MediaType.TEXT_XML})
    @Wrapped(element="pastries")
    public List<Pastry> getPastries() {
        return PastryRepository.getPastries();
    }
```
**Expected behavior**

When calling my REST endpoint, I expect something like: 

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<pastries>
   <pastry><name>Tartelette Fraise</name><price>2.0</price><size>S</size><status>available</status></pastry>
   <pastry><name>Eclair Cafe</name><price>2.5</price><size>M</size><status>available</status></pastry>
   <pastry><name>Millefeuille</name><price>4.4</price><size>L</size><status>available</status></pastry>
</pastries>
```

and that's the result I have when testing in JVM mode.

**Actual behavior**

When running in native mode, I have the following result:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><pastries/>
```

When calling REST endpoint that returns single entity (ex: `/pastry/{name}`), JAXB serialization works as expected even in native mode.

**To Reproduce**
1. Create a `Pastry` bean annotated with `@RegisterForReflection` and `@XmlRootElement`
2. Create a JAX-RS `PastryResource` with a method annotated with `@Wrapped(element="pastries")` and returning a List of Pastry
3. Compile your app in native mode and invoke the JAX-RS endpoint.

I have a demo project here: https://github.com/microcks/api-lifecycle/tree/master/api-pastry-demo/api-implementations/quarkus-api-pastry but it already contains the workaround I found (see below). You will have to remove it first for reproducing the failure.

**Configuration**

No properties added.

**Environment:**
 - Output of `uname -a` or `ver`: `Darwin lbroudou1-mac 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64`
 - Output of `java -version`: `OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.8+10)`
 - GraalVM version (if different from Java): used the docker image for native compilation
 - Quarkus version or git rev: `1.7.1.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): `Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)`

**Workaround**

Found a workaround by adding a `reflection-config.json` file and the adhoc native compilation flag. Once added the annotation is well managed in native mode and the serialization works as expected.

```json
[
  {
    "name" : "org.jboss.resteasy.plugins.providers.jaxb.JaxbCollection",
    "allDeclaredConstructors" : true,
    "allPublicConstructors" : true,
    "allDeclaredMethods" : true,
    "allPublicMethods" : true,
    "allDeclaredFields" : true,
    "allPublicFields" : true
  },
  {
    "name" : "javax.xml.bind.annotation.W3CDomHandler",
    "allDeclaredConstructors" : true,
    "allPublicConstructors" : true,
    "allDeclaredMethods" : true,
    "allPublicMethods" : true,
    "allDeclaredFields" : true,
    "allPublicFields" : true
  }
]
```

I think that this setup should be managed within `resteasy-jaxb` extension as this is a common use-case in RestEasy. If that makes sense for you, I could contribute a PR.

### PR DESCRIPTION
**Pull request by lbroudoux**
This PR fixes #11806

It adds a `BuildStep` for checking if specific RestEasy JAXB annotations are used for enforcing the Xml of the collection wrapping element (see https://docs.jboss.org/resteasy/docs/3.6.1.Final/userguide/html_single/#JAXB_Collections). If found, it is adding the wrapping RestEasy implementations as well as `W3CDomHandler` to the reflective classes.

### CODE REVIEW

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (gsmet) [inline extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java:41]: Maybe we could be a bit more clever and exit the for loop so that it's done only once?

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (gsmet) [inline extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java:40]: Are the providers automatically added to RESTEasy somehow? If so I'm surprised they are not added for reflection there.

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks!

I added a suggestion and a question.

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (gsmet) [inline extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java:28]: Also, could you make it a parameter of the build step rather than injecting it here? It helps keeping track of the dependencies.

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (lbroudoux) [reply extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java:41]: Absolutely! Will change that.

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (lbroudoux) [reply extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java:28]: Ok, will change that if it's a best practice.

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (lbroudoux) [reply extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java:40]: Not sure to see what you mean... This solely based on experiment ;-) I do not have specific knowledge of RestEasy internal. Do you mean there's some other provider classes that should be required too ?

File: extensions/resteasy-jaxb/deployment/src/test/java/io/quarkus/resteasy/jaxb/deployment/ProducesXMLTestCase.java
Comment (gsmet) [inline extensions/resteasy-jaxb/deployment/src/test/java/io/quarkus/resteasy/jaxb/deployment/ProducesXMLTestCase.java:3]: I missed that one but we usually avoid star imports.

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (gsmet) [reply extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java]: I'm just a bit surprised that just registering them for reflection can make things work.

I would have expected either it would work as is or adding them for reflection won't be enough.

**Comment by gsmet** [comment]
I you can adjust the star import and squash everything into one commit, I think we're good to go.

File: extensions/resteasy-jaxb/deployment/src/test/java/io/quarkus/resteasy/jaxb/deployment/ProducesXMLTestCase.java
Comment (lbroudoux) [reply extensions/resteasy-jaxb/deployment/src/test/java/io/quarkus/resteasy/jaxb/deployment/ProducesXMLTestCase.java:3]: Will do.

File: extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java
Comment (lbroudoux) [reply extensions/resteasy-jaxb/deployment/src/main/java/io/quarkus/resteasy/jaxb/deployment/ResteasyJaxbProcessor.java]: Yes just work as is ;-)

**Review by gsmet — APPROVED** [review]
Looks good, thanks. I will wait for CI before merging!

**Pull request merged**
